repos/fdbt-site/cypress_tests/cypress/support/steps.ts

```diff
@@ -406,35 +426,107 @@ export const editPassengerTypeOtherProductsPage = () => {
     clickElementByText('Back');
 };
 
-export const editStartDateOtherProductsPage = () => {
+export const editPassengerTypeOtherProductsPage = () => {
     clickRandomElementInTable('govuk-table__body', 'product-link');
     getElementById('product-name').should('not.be.empty');
     getElementById('product-status').should('not.be.empty');
     getElementById('fare-type').should('not.be.empty');
+    editPassengerType();
+};
+
+export const editPassengerTypePointToPointPage = () => {
+    cy.get('table tbody tr:has(td:nth-child(2):contains("1"))')
+        .invoke('index')
+        .then((i) => {
+            cy.wrap(i).as('index');
+        });
+    cy.get('@index').then((index) => {
+        cy.log(index.toString());
+        cy.get('table tbody tr:has(td:nth-child(2):contains("1"))')
+            .find(`[id=service-link-${index}]`)
+            .click()
+            .then(() => {
+                cy.get('[class=govuk-table__body]').find('a').click();
+                getElementById('service-name').should('not.be.empty');
+                getElementById('service-status').should('not.be.empty');
+                editPassengerType();
+            });
+    });
+};
+
+const editStartDate = () => {
     clickElementById('start-date-link');
     clearDates();
     let dateInput = completeProductDateInformationPage();
     cy.get('[id=start-date]').should('have.text', dateInput.startDate);
     clickElementByText('Back');
 };
 
-export const editEndDateOtherProductsPage = () => {
+export const editStartDateOtherProductsPage = () => {
     clickRandomElementInTable('govuk-table__body', 'product-link');
     getElementById('product-name').should('not.be.empty');
     getElementById('product-status').should('not.be.empty');
     getElementById('fare-type').should('not.be.empty');
+    editStartDate();
+};
+
+export const editStartDatePointToPointPage = () => {
+    cy.get('table tbody tr:has(td:nth-child(2):contains("1"))')
```

Comment:
is it possible for us to be using id selectors instead of these ones? Just because these seem quite fragile - if we changed the UI in that area, it would probably break these tests.

Reply:
i have attempted been unable to do so

Reply:
sorted